Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -161,4 +165,169 @@ instance : HasForget₂ ProfiniteGrp Grp where
     map := fun f => f.toMonoidHom
   }
 
+/-- A topological group that has a ContinuousMulEquiv to a profinite group is profinite. -/
+def ofContinuousMulEquivProfiniteGrp {G : ProfiniteGrp.{u}} {H : Type v} [TopologicalSpace H]
+    [Group H] [TopologicalGroup H] (e : ContinuousMulEquiv G H) : ProfiniteGrp.{v} :=
+  letI : CompactSpace H := Homeomorph.compactSpace e.toHomeomorph
+  letI : TotallyDisconnectedSpace G := Profinite.instTotallyDisconnectedSpaceαTopologicalSpaceToTop
+  letI : TotallyDisconnectedSpace H := Homeomorph.totallyDisconnectedSpace e.toHomeomorph
+  .of H
+
+/-- A closed subgroup of a profinite group is profinite. -/
+def ofClosedSubgroup {G : ProfiniteGrp}
```

Comment:
Did I remember wrongly or you defined the class of closed subgroups in another PR?

Reply:
yes, I defined it, maybe I should change the dependency.

Reply:
Well, if you have the class for closed subgroups, you should have `[ClosedSubgroup H]` as class variable.

Reply:
sorry that it is still only a structure, I think I need to learn about the APIs to work on it.

Reply:
No, it is me that was unclear, I am sorry. If you have a look at [Subgroup.Normal.mem_comm_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/Subgroup/Basic.html#Subgroup.Normal.mem_comm_iff) (I just picked up the first result about subgroups, there is nothing specific about this example), you see that once you have a class like `ClosedSubgroup` you can declare `H` to be a term of type `ClosedSubgroup G` instead of imposing that it is a subgroup _and_ it is closed. So the statement should look like 
```lean
def ofClosedSubgroup {G : ProfiniteGrp} (H : ClosedSubgroup G) : ProfiniteGrp :=...
```

Reply:
OK, I would rearrange the PRs as soon as possible.

Reply:
rearranged, it is now stated as you suggested